internal/databases/greenplum/extract_provider_with_spec.go

```diff
@@ -0,0 +1,65 @@
+package greenplum
+
+import (
+	"strings"
+
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/internal"
+	"github.com/wal-g/wal-g/internal/databases/postgres"
+)
+
+const (
+	aoSegNamespace = "pg_aoseg."
+)
+
+type RestoreDescMaker struct{}
+
+func (m RestoreDescMaker) Make(restoreParameters []string, names postgres.DatabasesByNames) (postgres.RestoreDesc, error) {
+	restoredDatabases, err := postgres.DefaultRestoreDescMaker{}.Make(restoreParameters, names)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, dbInfo := range names {
+		if _, dbNotSkipped := restoredDatabases[dbInfo.Oid]; !dbNotSkipped {
+			continue
+		}
+		for table, tableID := range dbInfo.Tables {
+			if m.FromAoSegNamespace(table) {
+				restoredDatabases.Add(dbInfo.Oid, tableID)
+			}
+		}
+	}
+
+	return restoredDatabases, nil
+}
+
+func (m RestoreDescMaker) FromAoSegNamespace(tableName string) bool {
+	return strings.HasPrefix(tableName, aoSegNamespace)
+}
+
+type ExtractProviderDBSpec struct {
+	*postgres.ExtractProviderDBSpec
```

Comment:
What is the goal of including this implementation? It only stores restoreParameters.

Reply:
It also hold `FilterFilesToUnwrap()` method that I call in overridden `Get()`. Maybe I should move `FilterFilesToUnwrap()` into RestoreDesc struct. I don't know.